dcc-portal-server/src/main/java/org/icgc/dcc/portal/server/resource/ui/UISearchResource.java

```diff
@@ -99,6 +99,8 @@ public Mutations getDonorMutations(
       @ApiParam(value = "From") @QueryParam("from") @DefaultValue("1") IntParam from,
       @ApiParam(value = "Size") @QueryParam("size") @DefaultValue("10") IntParam size) {
 
+    checkRequest(donorId == null, "donorId query param is required.");
+
```

Comment:
Yea you are right about `isNullOrEmpty`, and yea it absolutely can be null when you don't even specify `?donorId=` in the query string. No key, no object.